image/roxctl/Dockerfile.rhtap

```diff
@@ -0,0 +1,53 @@
+# TODO: can we follow tags?
+FROM registry.access.redhat.com/ubi8/go-toolset:1.19 as builder
+
+WORKDIR /go/src/github.com/stackrox/rox/app
+
+# TODO: do we want to avoid this by COPYing each required directory separately?
+COPY . .
+
+RUN git config --global --add safe.directory /go/src/github.com/stackrox/rox/app && \
+    # We could add tag fetching to: https://github.com/redhat-appstudio/build-definitions/blob/main/task/git-clone/0.1/git-clone.yaml
+    # TODO: if git fetch fails, check if privileges/file owners are OK, usually only on local build because repo might have moved further.
+    # TODO: this prevents hermetic builds
+    git fetch --tags --force && \
+    mkdir -p image/bin
+
+ENV CI=1 GOFLAGS="" GOTAGS="release"
+
+RUN RACE=0 CGO_ENABLED=1 GOOS=linux GOARCH=$(go env GOARCH) BUILD_TAG=$(make tag) scripts/go-build.sh ./roxctl && \
+    cp bin/linux_$(go env GOARCH)/roxctl image/bin/roxctl
+
+# TODO: can we follow tags?
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest as app
+
+COPY --from=builder /go/src/github.com/stackrox/rox/app/image/bin/roxctl /usr/bin/roxctl
+
+# TODO: this prevents hermetic builds
+RUN microdnf upgrade -y --nobest && \
```

Comment:
Do you know why this drops hermetic status? Is it because network requests to RPM repos are not considered kosher?

Reply:
The `buildah` task uses [this](https://github.com/redhat-appstudio/build-definitions/blob/main/task/buildah/0.1/buildah.yaml#L139) to ensure hermetic builds, i.e. no network is shared with the build process IIUC, so no network connections are possible.

Reply:
Well, that's now how CPaaS worked. Asked about this in Slack https://redhat-internal.slack.com/archives/C04PZ7H0VA8/p1696574824433919

Reply:
TLDR from Slack conversation: it is how it is right now, they are aware of the issue and working on a solution without a target date.